### ISSUE TITLE
Splashscreen / Input thread

### ISSUE THREAD

**Issue by Kangz** [Unvanquished/Unvanquished#354]
I'd be nice to have a splash screen while the game starts (6 sec on my SSD with hot OS cache). Also we will want inputs to be processed in parallel of the game to reduce latency and enable neat tricks. The two are linked since with SDL all the windowing function must be done in the main thread, including getting new events.

An idea of architecture it to have the game create a new "main2" thread as soon as it start. While the main2 thread does the actual initialization, the main thread shows a splash screen. When the game is ready to show the first frame, the splash screen is removed and the game window shown. In order to be able to use the SDL window functions from other threads, main can receive messages that contain callbacks to be executed. Basically main wakes up on every new message or each 10ms and process messages and SDL events (SDL events are put in a queue to be consumed by Com_Frame).

Kangz has a proof of concept implementation of the splash screen only.
- Mentor: Kangz
- Knowledge: multithreading, SDL, event loop
- Difficulty: 6

[Mentored projects on the Unv wiki](http://unvnet.net/wiki/index.php/Mentored_Coding_Project)

**Comment by bmorel** [comment]
I would be in favor to close this report and open one more focused on the input thread, as imo the splashscreen thing is fixed.
Also, just to be sure, can we have a list of systems (kernel + DE, and if possible the complete name of the OS instead of just kernel name+version) where «the window manager [...] tell the user “The application stopped responding”» ? Because to me, this looks like a windows-only issue, and while it's not a negligible issue, adding this information will help fixing the issue.

**Comment by illwieckz** [comment]
> Because to me, this looks like a windows-only issue, and while it's not a negligible issue, adding this information will help fixing the issue.

I get it with Ubuntu and GNOME every time.
